pkg/controller/adam/adam.go

```diff
@@ -11,26 +11,65 @@ import (
 	uuid "github.com/satori/go.uuid"
 	log "github.com/sirupsen/logrus"
 	"io/ioutil"
+	"net/url"
 	"os"
 	"path"
+	"strconv"
+	"strings"
 	"time"
 )
 
 type Ctx struct {
-	dir         string
-	url         string
-	serverCA    string
-	insecureTLS bool
-	AdamRemote  bool
+	dir              string
+	url              string
+	serverCA         string
+	insecureTLS      bool
+	AdamRemote       bool
+	AdamRemoteRedis  bool   //use redis for obtain logs and info
+	AdamRedisUrlEden string //string with redis url for obtain logs and info
+}
+
+//parseRedisUrl try to use string from config to obtain redis url
+func parseRedisUrl(s string) (addr, password string, databaseID int, err error) {
+	URL, err := url.Parse(s)
+	if err != nil || URL.Scheme != "redis" {
+		return "", "", 0, err
+	}
+
+	if URL.Host != "" {
+		addr = URL.Host
+	} else {
+		addr = "localhost:6379"
+	}
+	if URL.Path != "" {
+		if databaseID, err = strconv.Atoi(strings.Trim(URL.Path, "/")); err != nil {
+			return "", "", 0, err
+		}
+	} else {
+		databaseID = 0
+	}
+	password = URL.User.Username()
+	return
 }
 
-func (adam *Ctx) getLoader() loaders.Loader {
+//getLoader return loader object from Adam`s config
```

Comment:
Could you please explain a little. I understand that getLoader gets some Loader. But what is loader?

Reply:
In this function we get an implementation of Loader interface based on config for obtain logs/info from different sources (files, http stream, redis).